app/main.py

```diff
@@ -97,37 +99,68 @@ def grab_secrets(self):
 
         if secrets_keys is not None:
             for key_info in filter(None, secrets_keys.split(';')):
-                key_name, _, key_version = key_info.strip().partition(':')
-                _logger.info('Retrieving secret name:%s with version: %s', key_name, key_version)
+                key_name, key_version, cert_filename, key_filename = self._split_keyinfo(key_info)
+                _logger.info('Retrieving secret name:%s with version: %s output certFileName: %s keyFileName: %s', key_name, key_version, cert_filename, key_filename)
                 secret = client.get_secret(vault_base_url, key_name, key_version)
-                output_path = os.path.join(self._secrets_output_folder, key_name)
+                output_path = os.path.join(self._secrets_output_folder, cert_filename)
                 if secret.kid is not None:
                     _logger.info('Secret is backing certificate. Dumping private key and certificate.')
-                    self._dump_pfx(secret.value, key_name)
+                    self._dump_pfx(secret.value, cert_filename, key_filename)
+                elif (key_name != cert_filename):
+                    _logger.error('Cert filename provided for secret not backing a certificate.')
+                    return
```

Comment:
THe confition here is wrong - the cert_filename would match the key_filename if they did not give two file names for a secret that does not have two possible ones.  So the error would be that cert_filename != key_filename for that case.

This should throw an error as otherwise it still signals success even though it did not produce the output requested.  (Just like what would happen if it fails to write the file for some reason because open() failed)

The error message should tell you enough information to find which secret was incorrectly specified.  Please include the key_info text along with the error that it has dual filenames provided for something that is just one.  (That is, the secret produces just 1 file so it does not know what to do with the second filename)

Reply:
...unless you don't want any naming capability for the base secret, in which case the condition is but the error handling still needs the failure and clear indication of what failed.